src/steadystate_adjoint.jl

```diff
@@ -103,15 +102,18 @@ end
     else
         if linsolve === nothing && isempty(sensealg.linsolve_kwargs)
             # For the default case use `\` to avoid any form of unnecessary cache allocation
```

Comment:
Yeah I don't know about that comment. I think it's just old. (a) `\` always allocates because it uses `lu` instead of `lu!`, so it's re-allocating the while matrix which is larger than any LinearSolve allocation, and (b) we have since 2023 setup tests on StaticArrays, so the immutable path is non-allocating. I don't think (b) was true when this was written.

Reply:
So glad we can remove this branch altogether.